core/src/main/scala/akka/persistence/cassandra/journal/CassandraTagRecovery.scala

```diff
@@ -58,13 +59,10 @@ trait CassandraTagRecovery {
     rawEvent
   }
 
-  private[akka] def sendTagProgress(tp: Map[Tag, TagProgress], ref: ActorRef): Future[Done] = {
+  private[akka] def sendTagProgress(pid: String, tp: Map[Tag, TagProgress], ref: ActorRef): Future[Done] = {
     implicit val timeout = Timeout(10.second)
     log.debug("Recovery sending tag progress: {}", tp)
-    val progressSets = tp.map {
-      case (tag, progress) => (ref ? SetTagProgress(tag, progress)).mapTo[TagWriter.SetTagProgressAck.type]
-    }
-    Future.sequence(progressSets).map(_ => Done)
+    (ref ? PidRecovering(pid, tp)).mapTo[PidRecoveringAck.type].map(_ => Done)
```

Comment:
the mapTo is not really needed, but good for documenting the types of the interaction

Reply:
Looking forward to re-writing this in typed :)